### PR TITLE
initial commit for cli

### PR DESCRIPTION
**Pull request by edgarriba**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by shijianjian** [comment]
I think we may have a KorniaModel module that has a to_onnx function, then we make it like 'YuNet().to_onnx()'. 

I was thinking that the CLI may do some syntax sugar such as 'kornia-cli rgb2gray/YuNet IMAGE-PATH OUTPUT-PATH'.

Maybe we can also add a process chain such as 'kornia-cli rgb2gray->resize<256,256>->YuNet IMAGE-PATH OUTPUT-PATH'.

Probably then we can have some user-defined chain like 'kornia-cli chain edit'. It will then pop up a VIM of JSON that contains user defined chains. 


Just some random thoughts.

**Comment by edgarriba** [comment]
@shijianjian
> I think we may have a KorniaModel module that has a to_onnx function, then we make it like 'YuNet().to_onnx()'.

why don't we extend the `ImageModule` ?

> I was thinking that the CLI may do some syntax sugar such as 'kornia-cli rgb2gray/YuNet IMAGE-PATH OUTPUT-PATH'.

yes, but i would like to have different commands layers: `export` (to export to onnx, tensorrt, openvino, etc), `compute` (mostly inference models and run scripts`, `serve` (start a fastapi service and expose minimum apis, like to connect from a camera, or models via grpc/proto (json is too expensive to images/videos)

> Maybe we can also add a process chain such as 'kornia-cli rgb2gray->resize<256,256>->YuNet IMAGE-PATH OUTPUT-PATH'.

I'd call it `pipe` -- > `kornia-cli pipe src=webcam ! rgb2gray ! facedetect model=yunet nms=0.8 ! sink=rerun`
Which internally can use Limbus, and advanced version can use rust.

File: pyproject.toml [hunk 1]
```diff
@@ -1,21 +1,13 @@
 [build-system]
 build-backend = "setuptools.build_meta"
-requires = [
-  "setuptools>=61.2",
-]
+requires = [ "setuptools>=61.2" ]
```
Comment (johnnv1) [inline pyproject.toml:3]: some unrelated to the patch. We can remove the pyproject formater from pre-commit and fix this file. It has been updated some time ago splitting some things (like ruff configs) unnecessary -- could be in another PR

File: kornia/cli.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+"""Command-line interface for Kornia."""
```
Comment (johnnv1) [inline kornia/cli.py:1]: personally, i would go with 

```console
foo@bar:/tmp/kornia $ tree kornia
kornia/
├── augmentation
│   ├── ...
├── ...
├── commands
│   ├── main.py // principal file with the principal CLI config
│   ├── ...
│   └── exporter.py // file with all export-related functions
├── ...
```

I can reorganize to have something like this above, in a further PR, if it makes sense for you

here is an example of it https://github.com/johnnv1/CCAgT-utils/tree/first-release/CCAgT_utils/commands

File: kornia/cli.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    if config.output_path.exists():
+        logger.info("Model exported to %s", config.output_path)
+    else:
+        logger.error("Error exporting model to %s", config.output_path)
+
+
+def export_onnx_model_config_resolver(model_name: str, output_path: Path) -> ExportOnnxModelConfig:
+    """Resolve the configuration for exporting a model to ONNX format."""
+    if model_name == "kornia/yunet":
```
Comment (johnnv1) [inline kornia/cli.py:47]: i would assume, any model without the explicit module is from kornia
```suggestion
    if model_name in {"kornia/yunet", "yunet"}:
```

File: kornia/cli.py (hunk 2 above)
Comment (edgarriba) [reply kornia/cli.py:1]: yes, why not. My only question now is whether this should be packaged with the libirary, or where exactly should live, plus what's parts should be an api (like the KorniaModule that Jian suggested)

File: kornia/cli.py (hunk 3 above)
Comment (edgarriba) [reply kornia/cli.py:47]: once matured, this should be an enum to ve less error prone

File: pyproject.toml (hunk 1 above)
Comment (edgarriba) [reply pyproject.toml:3]: +1

**Comment by edgarriba** [comment]
will be somehow part of #3013
